ferry/database/sync_db_courses.py

```
@@ -812,6 +830,34 @@ def sync_db_courses(
             commit_deletions(table_name, deleted_rows, conn)
         print("\033[F", end="")
 
+        changed_course_ids = list(course_sync_diff_by_id.keys())
+        if changed_course_ids:
+            ids_sql = ",".join(str(int(cid)) for cid in changed_course_ids)
+            conn.execute(
+                text(
+                    f"UPDATE courses SET last_sync_diff = NULL "
+                    f"WHERE last_sync_diff IS NOT NULL AND course_id NOT IN ({ids_sql})"
```

Ruff (0.15.9)

[error] Possible SQL injection vector through string-based query construction

(S608)